pika/connection.py

```diff
@@ -1102,14 +1149,15 @@ def _deliver_frame_to_channel(self, value):
 
         """
         if not value.channel_number in self._channels:
-            if self._is_basic_deliver_frame(value):
-                self._reject_out_of_band_delivery(value.channel_number,
-                                                  value.method.delivery_tag)
-            else:
-                LOGGER.warning("Received %r for non-existing channel %i", value,
-                               value.channel_number)
+            # This should never happen and would constitute breach of the
+            # protocol
+            LOGGER.critical(
+                'Received %s frame for unregistered channel %i on %s',
+                value.NAME, value.channel_number, self)
             return
-        return self._channels[value.channel_number]._handle_content_frame(value)
+
+        # pylint: disable=W0212
+        self._channels[value.channel_number]._handle_content_frame(value)
```

Comment:
REVIEWME `_handle_content_frame` doesn't return anything explicit, and neither does `_deliver_frame_to_channel`